cpp/dolfinx/fem/DirichletBC.h

```diff
@@ -469,18 +470,29 @@ class DirichletBC
     return {_dofs0, _owned_indices0};
   }
 
-  /// Set bc entries in `x` to `scale * x_bc`
+  /// @brief Set entries in an array that are constrained by Dirichlet
+  /// boundary conditions.
   ///
-  /// @param[in] x The array in which to set `scale * x_bc[i]`, where
-  /// x_bc[i] is the boundary value of x[i]. Entries in x that do not
-  /// have a Dirichlet condition applied to them are unchanged. The
-  /// length of x must be less than or equal to the index of the
-  /// greatest boundary dof index. To set values only for
-  /// degrees-of-freedom that are owned by the calling rank, the length
-  /// of the array @p x should be equal to the number of dofs owned by
-  /// this rank.
-  /// @param[in] scale The scaling value to apply
-  void set(std::span<T> x, T scale = 1) const
+  /// Entries in the array `x` that are constrained by a Dirichlet
+  /// boundary conditions are set to `alpha * (x_bc - x0)`, where `x_bc`
+  /// is the boundary condition value.
+  ///
+  /// For elements with point-wise evaluated degrees-of-freedom, e.g.
```

Comment:
I made this more precise and then more direct about what the user should be aware of.

`x_bc` is the value of the boundary condition interpolated into the finite element space via the degrees of freedom. Note that for many finite elements, the degrees of freedom are not straightforward point evaluations, as for the Lagrange element.